### PR TITLE
Documentation for stakeholders

### PR DESCRIPTION
**Pull request by g4s8**
#1310 - added documentation for some stakeholders

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1409](https://github.com/zerocracy/farm/issues/1409) is now in scope, role is `REV`

**Comment by g4s8** [comment]
@0crat assign @amihaiemil

**Comment by 0crat** [comment]
> @0crat assign @amihaiemil  ([here](https://github.com/zerocracy/farm/issues/1409#issuecomment-404153946))

@g4s8 This pull request [#1409](https://github.com/zerocracy/farm/issues/1409) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1921238863); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Manual assignment of issues is discouraged, see [§19](http://www.zerocracy.com/policy.html#19): -5 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

File: README.md [hunk 1]
```diff
@@ -277,6 +277,8 @@ daylight saving).
 Just fork it, make changes, run `mvn clean install -Pqulice,codenarc`,
 and submit a pull request. Read
 [this](http://www.yegor256.com/2014/04/15/github-guidelines.html), if lost.
+Don't forget to add documentation for groovy scripts if create new
+stakeholder.
```
Comment (amihaiemil) [inline README.md]: @krzyk "Don't forget to add GroovyDocs if **you** create new stakeholders."

File: src/main/groovy/com/zerocracy/stk/pm/comm/notify.groovy [hunk 2]
```diff
@@ -22,6 +22,14 @@ import com.zerocracy.farm.Assume
 import com.zerocracy.Project
 import com.zerocracy.pm.ClaimIn
 
+/**
+ * Generic notify stakeholder. It can understand what channel
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/comm/notify.groovy]: @g4s8 I would change ``Generic notify stakeholder`` to ``Stakeholder for notifications``.

File: src/main/groovy/com/zerocracy/stk/pm/comm/notify.groovy [hunk 3]
```diff
@@ -22,6 +22,14 @@ import com.zerocracy.farm.Assume
...
+ * to use to notify by token (claim parameter)
+ * and produce more concrete notification claim.
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/comm/notify.groovy]: @g4s8 "and produce **a** more concrete notification claim.""

File: src/main/groovy/com/zerocracy/stk/pm/comm/notify_all.groovy [hunk 4]
```diff
@@ -25,6 +25,13 @@ import com.zerocracy.pm.ClaimIn
 import com.zerocracy.pmo.Awards
 import com.zerocracy.pmo.People
 
+/**
+ * Notify all users. Claim sender can specify minimum required reputation
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/comm/notify_all.groovy]: @g4s8 "Notify all users" about what? Can you give an example? This seems like a stakeholder which broadcasts a message :D

File: src/main/groovy/com/zerocracy/stk/pm/cost/funding/contribute_by_stripe.groovy [hunk 5]
```diff
@@ -25,6 +25,14 @@ import com.zerocracy.farm.Assume
 import com.zerocracy.pm.ClaimIn
 import com.zerocracy.pm.cost.Ledger
 
+/**
+ * This stakeholder reacts to user contribution events.
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/cost/funding/contribute_by_stripe.groovy]: @g4s8 In what way does it "react", what does it do?

File: src/main/groovy/com/zerocracy/stk/pm/cost/funding/recharge_on_deficit.groovy [hunk 6]
```diff
@@ -26,6 +26,14 @@ import com.zerocracy.pm.cost.Estimates
 import com.zerocracy.pm.cost.Ledger
 import com.zerocracy.pmo.recharge.Recharge
 
+/**
+ * This stakeholder automatically recharge the project if detects
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/cost/funding/recharge_on_deficit.groovy]: @g4s8 "This stakeholder automatically recharge**s** the project if **it** detects a deficit. It uses **the** same Stripe account which was used previously **to** fund the project".

Question: does this stakeholder always run? What if the PO doesn't want this automatic recharging?

File: src/main/groovy/com/zerocracy/stk/pm/cost/funding/remember_stripe_charge.groovy [hunk 7]
```diff
@@ -24,6 +24,13 @@ import com.zerocracy.farm.Assume
 import com.zerocracy.pm.ClaimIn
 import com.zerocracy.pmo.recharge.Recharge
 
+/**
+ * Stakeholder to save Stripe customer account
+ * to be able to recharge this project automatically on deficit.
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/cost/funding/remember_stripe_charge.groovy]: @g4s8 "Stakeholder **which** saves the customer's Stripe account, in order to be able to recharge **the** project atuomatically on deficit"

BTW, we are **not** allowed to store the customer's credit card data! I hope that's not what we store :D

File: src/main/groovy/com/zerocracy/stk/pm/cost/funding/stop_recharging.groovy [hunk 8]
```diff
@@ -23,6 +23,13 @@ import com.zerocracy.farm.Assume
 import com.zerocracy.pm.ClaimIn
 import com.zerocracy.pmo.recharge.Recharge
 
+/**
+ * Stop auto-recharge for project. The project will not be funded
+ * automatically from saved Stripe account until it funded manually.
```
Comment (amihaiemil) [inline src/main/groovy/com/zerocracy/stk/pm/cost/funding/stop_recharging.groovy]: @g4s8 "until it **is** funded manually"

Does this mean that the PO can stop automatic recharging, but the mechanism will start again once they manually fund the project?

File: src/main/groovy/com/zerocracy/stk/pm/comm/notify_all.groovy (hunk 4 above)
Comment (g4s8) [reply src/main/groovy/com/zerocracy/stk/pm/comm/notify_all.groovy]: @amihaiemil yes, it is. This stakeholder broadcasts a message to all users

File: src/main/groovy/com/zerocracy/stk/pm/cost/funding/recharge_on_deficit.groovy (hunk 6 above)
Comment (g4s8) [reply src/main/groovy/com/zerocracy/stk/pm/cost/funding/recharge_on_deficit.groovy]: > does this stakeholder always run? What if the PO doesn't want this automatic recharging?

@amihaiemil PO can pause the project with `0crat pause` in Slack channel

File: src/main/groovy/com/zerocracy/stk/pm/cost/funding/remember_stripe_charge.groovy (hunk 7 above)
Comment (g4s8) [reply src/main/groovy/com/zerocracy/stk/pm/cost/funding/remember_stripe_charge.groovy]: > we are not allowed to store the customer's credit card data

@amihaiemil we just store Stripe customer id which can be used to recharge via API

File: src/main/groovy/com/zerocracy/stk/pm/cost/funding/stop_recharging.groovy (hunk 8 above)
Comment (g4s8) [reply src/main/groovy/com/zerocracy/stk/pm/cost/funding/stop_recharging.groovy]: > Does this mean that the PO can stop automatic recharging, but the mechanism will start again once they manually fund the project?

@amihaiemil yes

**Comment by g4s8** [comment]
@amihaiemil I fixed grammar and answered questions, please see

**Comment by amihaiemil** [comment]
@g4s8 Thanks, good to merge :)

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1409](https://github.com/zerocracy/farm/issues/1409) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +31 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**